### PR TITLE
5429: Other Filers Contact entity

### PR DESCRIPTION
**Pull request by sutt0n**

### CODE REVIEW

File: shared/src/business/entities/cases/Case.js [hunk 1]
```diff
@@ -545,6 +548,15 @@ Case.VALIDATION_RULES = {
...
     .description('Reminder for clerks to review the Order to Show Cause.'),
+  otherFilers: joi
+    .array()
+    .items(joi.object().meta({ entityName: 'OtherFilerContact' }))
+    .unique(
+      (a, b) =>
+        a.otherFilerType === UNIQUE_OTHER_FILER_TYPE &&
+        b.otherFilerType === UNIQUE_OTHER_FILER_TYPE,
+    )
+    .description('List of OtherFilerContact Entities for the case.'),
```
Comment (matthopson) [inline shared/src/business/entities/cases/Case.js]: Should we add optional / required, or some other conditional validation?

**Review by kkoskelin — APPROVED** [review]

File: shared/src/business/entities/cases/Case.test.js [hunk 2]
```diff
@@ -211,6 +213,146 @@ describe('Case entity', () => {
...
+        },
+        {
+          additionalName: 'Second Other Filer',
+          address1: '1337 12th Ave',
+          city: 'Flavortown',
+          country: 'USA',
+          countryType: 'domestic',
+          email: 'mayor@flavortown.com',
+          name: 'Guy Fieri',
+          otherFilerType: UNIQUE_OTHER_FILER_TYPE,
```
Comment (matthopson) [inline shared/src/business/entities/cases/Case.test.js]: Just a small nit: sometimes in tests like this, I'll throw a little `// comment` next to the line noting _why_ it's going to cause a fail.

Comment (matthopson) [reply shared/src/business/entities/cases/Case.test.js]: Looking again, it's in the test desc, so ¯\_(ツ)_/¯

File: shared/src/business/entities/cases/Case.js (hunk 1 above)
Comment (sutt0n) [reply shared/src/business/entities/cases/Case.js]: `.optional()` is there by default, i believe, but i'll explicitly add it to `otherPetitioners` and this one (`otherFilers`) for context

File: shared/src/business/entities/contacts/OtherFilerContact.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * returns the constructor used for creating the OtherFilerContact entity
+ */
+exports.getOtherFilerContact = ContactFactory.createContactFactory({
+  additionalErrorMappings: {
+    otherFilerType: 'Select a filer type',
+  },
+  additionalValidation: {
+    otherFilerType: joi.string().valid(...OTHER_FILER_TYPES),
+  },
+});
```
Comment (matthopson) [inline shared/src/business/entities/contacts/OtherFilerContact.js:18]: Can we add a test for this entity?

**Review by rachelschneiderman — APPROVED** [review]

**Review by matthopson — APPROVED** [review]

**Pull request merged**